libs/k6/src/slack/slack.js

```diff
@@ -0,0 +1,89 @@
+import http from 'k6/http';
+
+import { createDefaultPayload } from './payload';
+import * as config from '../config';
+
+function performanceMetrics(data) {
+  const numberOfRequests = data.metrics.http_reqs.values['count'];
+  const maxThroughput = data.metrics.http_reqs.values['rate'].toFixed(2);
+  const avgReqDuration =
+    data.metrics.http_req_duration.values['avg'].toFixed(2);
+  const p95ReqDuration =
+    data.metrics.http_req_duration.values['p(95)'].toFixed(2);
+  const p99ReqDuration =
+    data.metrics.http_req_duration.values['p(99)'].toFixed(2);
+
+  return `*Number of Requests:* ${numberOfRequests} reqs
+*Max Throughput:* ${maxThroughput} reqs/s
+*Average Response Time:* ${avgReqDuration} ms
+*p(95) Response Time:* ${p95ReqDuration} ms
+*p(99) Response Time:* ${p99ReqDuration} ms`;
+}
+
+function checkMetrics(data) {
+  let numberOfChecks = data.root_group.checks.length;
+  let checksString = '';
+  let checks = data.root_group.checks;
+
+  for (let i = 0; i < numberOfChecks; i++) {
+    let passingChecks = checks[i].passes;
+    let totalChecks = checks[i].passes + checks[i].fails;
+    let successRate = (passingChecks / totalChecks) * 100;
+    checksString += `*Check*: '${checks[i].name}' had a Success Rate of: ${successRate}%\n`;
+  }
+  return checksString;
+}
+
+function buildPayload(data, reportType) {
+  var payload = createDefaultPayload();
+  let sectionBlocks = payload.attachments.find(
+    (attachments) => attachments.blocks[1].type === 'section',
+  );
+
+  switch (reportType) {
+    case 'performance':
+      sectionBlocks.blocks[1].text.text = performanceMetrics(data);
+      break;
+    case 'checks':
+      sectionBlocks.blocks[1].text.text = checkMetrics(data);
+      break;
+    default:
+      sectionBlocks.blocks[1].text.text = performanceMetrics(data);
+  }
+
+  sectionBlocks.blocks[3].elements[0].url = `${config.grafanaBaseUrl}/${
+    config.k6PrometheusDashboard
+  }\
+?orgId=1\
+&var-DS_PROMETHEUS=${config.datasource}\
+&var-namespace=${__ENV.NAMESPACE}\
```

Comment:
I will need to inject these by default in the generator code.
TODO: Might be worth it to see if I can get the start time of the test from K6.